haystack/preview/components/routers/conditional_router.py

```diff
@@ -0,0 +1,281 @@
+                )
+                raise RouteConditionException(error_message)
+
+            input_names.update(output_vars)
+
+            # Also add any additional variable that might be used within a "condition" expression.
+            ast = env.parse(route["condition"])
+            input_names.update(meta.find_undeclared_variables(ast))
+
+            # output_vars is a set guaranteed to have 0 or 1 elements
+            output_var = output_vars.pop() if output_vars else None
+            output_name = route.get("output_name", output_var)
+            # set proper output as well
+            output_types.update({output_name: route["output_type"]})
+
+        component.set_input_types(self, **{var: Any for var in input_names})
+        component.set_output_types(self, **output_types)
+
+    def to_dict(self) -> Dict[str, Any]:
+        for route in self.routes:
+            # output_type needs to be serialized to a string
+            route["output_type"] = serialize_type(route["output_type"])
+
+        return default_to_dict(self, routes=self.routes)
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "ConditionalRouter":
+        init_params = data.get("init_parameters", {})
+        routes = init_params.get("routes")
+        for route in routes:
+            # output_type needs to be deserialized from a string to a type
+            route["output_type"] = deserialize_type(route["output_type"])
+        return default_from_dict(cls, data)
+
+    def run(self, **kwargs):
+        """
+        Executes the routing logic by evaluating the specified boolean condition expressions
+        for each route in the order they are listed. The method directs the flow
+        of data to the output specified in the first route whose expression
+        evaluates to True. If no route's expression evaluates to True, an exception
+        is raised.
+
+        :param kwargs: A dictionary containing the pipeline variables, which should
+                   include all variables used in the "condition" templates.
+
+        :return: A dictionary containing the output and the corresponding result,
+             based on the first route whose expression evaluates to True.
+
+        :raises NoRouteSelectedException: If no route's expression evaluates to True.
+        """
+        # Create a Jinja native environment evaluate the condition templates
+        env = NativeEnvironment()
+
+        for route in self.routes:
+            try:
+                t = env.from_string(route["condition"])
+                if t.render(**kwargs):
+                    # if optional field output_name is not provided, use mandatory output
+                    # but since output is jinja expression, we need to strip the curly braces
+                    t_output = env.from_string(route["output"])
+                    output = t_output.render(**kwargs)
+
+                    # output_vars is a set guaranteed to have 0 or 1 elements
```

Comment:
```suggestion
                    # In case `output_name` was not specified, we use the name of the variable contained
                    # in `output` to address the appropriate output socket for this component.
```